### ISSUE TITLE
Running nx reset on Windows results in an error when attempting to delete nx-win32-x64-msvc.node

### ISSUE THREAD

**Issue by CreativeTechGuy** [nrwl/nx#23224]
### Current Behavior

This is almost identical to this bug: https://github.com/nrwl/nx/issues/20346 but after the location of that file was moved from `node_modules` to `.nx/cache`, the command which causes the issue changed from `npm ci` to `nx reset` (which attempts to delete the cache folder, effectively the same root cause as npm ci deleting node modules).

### Expected Behavior

No crash

### Steps to Reproduce

See this issue: https://github.com/nrwl/nx/issues/20346, just swap out `npm ci` for `nx reset`

### Nx Report

```shell
Node   : 20.9.0
OS     : win32-x64
npm    : 10.1.0
 
nx                 : 18.3.3
@nx/js             : 18.3.3
@nx/jest           : 18.3.3
@nx/linter         : 18.3.3
@nx/eslint         : 18.3.3
@nx/workspace      : 18.3.3
@nx/devkit         : 18.3.3
@nx/esbuild        : 18.3.3
@nx/eslint-plugin  : 18.3.3
@nx/plugin         : 18.3.3
@nrwl/tao          : 18.3.3
@nx/vite           : 18.3.3
typescript         : 5.4.5
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by CreativeTechGuy** [comment]
Tagging @MaxKless as requested.

**Comment by CreativeTechGuy** [comment]
This is also present in NX 19.0.1

**Comment by ChoSeoHwan** [comment]
Are you using webstorm?
I also had the same problem when using webstorm.

In webstorm, if you activate the eslint option, eslint is always running.
Also, by default, when configured through `nx init`, eslint is configured to use the `@nx/eslint-plugin` plugin.

From what I have confirmed, `@nx/eslint-plugin` uses `nx daemon`, which causes it to continue to use `nx-win32-x64-msvc.node`, making it impossible to delete the file.

Currently, @nx/eslint-plugin has been removed from the plugin settings in eslint.
Afterwards, it was confirmed that `nx reset` operates normally.

However, removing `@nx/eslint-plugin` is not the correct solution, and an official solution is needed for this.

**Comment by MaxKless** [comment]
Hey! Sorry, I haven't been able to repro this yet but will look into it soon.
@ChoSeoHwan are you on the latest versions? Can you paste your `nx report` & the errors you're seeing so we can confirm this is the same error?

**Comment by CreativeTechGuy** [comment]
@ChoSeoHwan, No, VS Code, but it sounds similar since I believe the root cause is the VS Code ESLint extension which works in a similar way.

**Comment by sshelake25** [comment]
This might take a few minutes.
Error: EPERM: operation not permitted, unlink '\\?\D:\a\1\s\.nx\cache\18.3.4-nx.win32-x64-msvc.node'

    at unlinkSync (node:fs:1883:3)

    at _unlinkSync (node:internal/fs/rimraf:214:14)
 NX   Daemon Server - Stopped
    at fixWinEPERMSync (node:internal/fs/rimraf:306:5)

I am facing same issue in azure devops CI pipeline

**Comment by MaxKless** [comment]
@sshelake25 please run `nx report` results and edit your comment with the results so I can see what version you're on. This bug should be fixed on the latest 18 or 19 versions.

**Comment by ChoSeoHwan** [comment]
> Hey! Sorry, I haven't been able to repro this yet but will look into it soon. @ChoSeoHwan are you on the latest versions? Can you paste your `nx report` & the errors you're seeing so we can confirm this is the same error?

### `nx report`
```
nx                 : 19.0.1
@nx/js             : 19.0.1
@nx/jest           : 19.0.1
@nx/eslint         : 19.0.1
@nx/workspace      : 19.0.1
@nx/devkit         : 19.0.1
@nx/esbuild        : 19.0.1
@nx/eslint-plugin  : 19.0.1
@nx/nest           : 19.0.1
@nx/node           : 19.0.1
@nx/plugin         : 19.0.1
@nx/webpack        : 19.0.1
nx-cloud           : 18.0.1
typescript         : 5.4.5
---------------------------------------
Community plugins:
@jscutlery/semver : 5.2.2
```

### Error reproduce
`.eslint.json`
```json
  "plugins": [
    "@nx"
  ],
```

`bash`
```bash
Error: EPERM: operation not permitted, unlink '[rootpath]\.nx\cache\19.0.1-nx.win32-x64-msvc.node'
    at Object.unlinkSync (node:fs:1874:11)
    at NodeFS.unlinkSync ([rootpath]\.pnp.cjs:39086:24)
    at makeCallSync.subPath.subPath ([rootpath]\.pnp.cjs:39718:26)
    at ZipOpenFS.makeCallSync ([rootpath]\.pnp.cjs:39935:14)
    at ZipOpenFS.unlinkSync ([rootpath]\.pnp.cjs:39717:17)
    at VirtualFS.unlinkSync ([rootpath]\.pnp.cjs:38717:24)
    at PosixFS.unlinkSync ([rootpath]\.pnp.cjs:38717:24)
    at NodePathFS.unlinkSync ([rootpath]\.pnp.cjs:38717:24)
    at _unlinkSync (node:internal/fs/rimraf:214:14)
    at fixWinEPERMSync (node:internal/fs/rimraf:306:5) {
  errno: -4048,
  code: 'EPERM',
  syscall: 'unlink',
  path: '[rootpath]\\.nx\\cache\\19.0.1-nx.win32-x64-msvc.node'
}

```


### remove nx eslint plugin (webstorm restart)
`.eslint.json`
```json
  "plugins": [],
```

`bash`
```bash
$ yarn nx reset

 NX   Resetting the Nx workspace cache and stopping the Nx Daemon.

This might take a few minutes.


 NX   Daemon Server - Stopped


 NX   Successfully reset the Nx workspace.

```

**Comment by MaxKless** [comment]
Thanks that's super valuable information. 
We're discussing potential fixes & will release something soon.

**Comment by sshelake25** [comment]
@MaxKless  Does anyone know which is nx version which does not have this nx reset issue ?

**Comment by sshelake25** [comment]
$ nx reset

 >  NX   Its time to update Nx 🎉

   Your repository uses a higher version of Nx (19.0.3) than your global CLI version (16.8.1)
   For more information, see https://nx.dev/more-concepts/global-nx


 NX   Resetting the Nx workspace cache and stopping the Nx Daemon.

This might take a few minutes.


 NX   Daemon Server - Stopped

nx reset

Clears all the cached Nx artifacts and metadata about the workspace and shuts down the Nx Daemon.

Options:
  --help     Show help                                                                                  [boolean]  --version  Show version number                                                                        [boolean]
Error: EPERM: operation not permitted, unlink 'D:\who\Omni\.nx\cache\19.0.3-nx.win32-x64-msvc.node'
    at Object.unlink (node:internal/fs/sync:93:18)
    at unlinkSync (node:fs:1853:17)
    at _unlinkSync (node:internal/fs/rimraf:214:14)
    at fixWinEPERMSync (node:internal/fs/rimraf:306:5)
    at rimrafSync (node:internal/fs/rimraf:200:14)
    at node:internal/fs/rimraf:253:9
    at Array.forEach (<anonymous>)
    at _rmdirSync (node:internal/fs/rimraf:250:7)
    at fixWinEPERMSync (node:internal/fs/rimraf:304:5)
    at rimrafSync (node:internal/fs/rimraf:200:14) {
  errno: -4048,
  code: 'EPERM',
  syscall: 'unlink',
  path: 'D:\\who\\Omni\\.nx\\cache\\19.0.3-nx.win32-x64-msvc.node'
}


$ nx report 

 >  NX   Its time to update Nx 🎉

   Your repository uses a higher version of Nx (19.0.3) than your global CLI version (16.8.1)
   For more information, see https://nx.dev/more-concepts/global-nx


 NX   Report complete - copy this into the issue template

Node   : 20.9.0
OS     : win32-x64
npm    : 10.1.0

nx (global)        : 16.8.1
nx                 : 19.0.3
@nx/js             : 19.0.3
@nx/linter         : 19.0.3
@nx/eslint         : 19.0.3
@nx/workspace      : 19.0.3
@nx/angular        : 19.0.3
@nx/cypress        : 19.0.3
@nx/devkit         : 19.0.3
@nrwl/devkit       : 17.0.2
@nx/eslint-plugin  : 19.0.3
@nrwl/tao          : 19.0.3
@nx/web            : 19.0.3
@nx/webpack        : 19.0.3
typescript         : 5.4.2
---------------------------------------
Community plugins:
@ngrx/effects          : 17.2.0
@ngrx/schematics       : 17.2.0
@ngrx/store            : 17.2.0
@ngrx/store-devtools   : 17.2.0
@ngx-formly/schematics : 6.3.0
@nx-dotnet/core        : 2.1.1
angular-calendar       : 0.31.0
ng-mocks               : 14.12.2
ngx-mailto             : 17.0.0
ngx-toastr             : 16.1.1
---------------------------------------
The following packages should match the installed version of nx
  - @nrwl/devkit@17.0.2

To fix this, run `nx migrate nx@19.0.3`
